kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/subject/TransportSubjectBuilderService.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.subject;
+
+import io.kroxylicious.proxy.authentication.TransportSubjectBuilder;
+
+public interface TransportSubjectBuilderService<C> extends AutoCloseable {
+    void initializeTransportSubjectBuilderService(C config);
+
+    TransportSubjectBuilder buildTransportSubjectBuilderService();
```

Comment:
The return type and the method name feel like they are at cross purposes. 

Shouldn't a `buildTransportSubjectBuilderService` return a `TransportSubjectBuilderService` ? 

Hang on why are there two different `TransportSubjectBuilderService` interfaces in this PR? We have this one in `io.kroxylicious.proxy.internal.subject` and one in ` io.kroxylicious.proxy.authentication`. The method signature disagreement applies equally to both...